paasta_tools/flink_tools.py

```diff
@@ -361,3 +409,443 @@ def get_flink_overview_from_paasta_api_client(
         service=service,
         instance=instance,
     )
+
+
+def get_flink_instance_details(
+    metadata: Mapping[str, Any],
```

Comment:
unfortunate that we don't have something fully typed like the k8s objects available to us here :(